### PR TITLE
Feature/delimiters prop

### PR DESCRIPTION
**Pull request by kevinch**
#### What is the purpose of this pull request?
Add the `prop` delimiters as well as some default.

#### What problem is this solving?
Hard coded delimiter `,` was quite restraining.

#### How should this be manually tested?
Checkout and test with file setup with different delimiters.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,4 +61,14 @@ render() {
...
+
+| Prop name        | Type  | Default      | Required | Description                                                             |
+| ---------------- | ----- | ------------ | -------- | ----------------------------------------------------------------------- |
+| `fileHeaders`    | array |              | true     | The headers (usually used by api) of the file. Order is crucial.        |
+| `onDataUploaded` | func  |              | true     | Callback function with the data as parameter. Null if the parse failed. |
+| `delimiters`     | array | `[";", ","]` | false    | The file will be tested with those characters to split the data.        |
+
+### _Delimiter rule_
+
+The component will estimate that if the amount of delimiters found is equal or above the `fileHeaders` lenght, then it will process the file. This does not guarantee a correct parsing but rather tries to add a small level of security.
```
Comment (klzns) [inline README.md]: Typo `length`

File: src/index.js [hunk 2]
```diff
@@ -12,59 +12,77 @@ import {
...
-      result = result.replace(/[\r\n\t]/g, ",")
+      delimiters.some((val, i) => {
+        const dynamicRegEx = new RegExp(delimiters[i], 'g')
+        const delimitersFound = (result.match(dynamicRegEx) || []).length
+
+        if (delimitersFound >= this.props.fileHeaders.length) {
+          this.handleResult(result, delimiters[i])
+          return true
+        } else if (i === delimiters.length - 1) {
+          this.props.onDataUploaded(null)
```
Comment (klzns) [inline src/index.js:34]: Is intended to be called multiple times?

**Review by klzns — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (kevinch) [reply src/index.js]: Yes, until the condition is found or we reach the end of the delimiters array.

Comment (kevinch) [reply src/index.js:34]: It can only be called once, at the end of the loop (delimiters' length).

Comment (klzns) [reply src/index.js:34]: Oh, my bad

**Review by klzns — APPROVED** [review]

**Pull request merged**